app/src/main/java/com/github/scribeWizTeam/scribewiz/Fragments/RecParameterFragment.kt

```diff
@@ -0,0 +1,247 @@
+package com.github.scribeWizTeam.scribewiz.Fragments
+
+import android.Manifest
+import android.annotation.SuppressLint
+import android.media.AudioFormat
+import android.media.AudioRecord
+import android.media.MediaPlayer
+import android.media.MediaRecorder
+import android.os.Bundle
+import android.os.CountDownTimer
+import android.util.Log
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import androidx.compose.foundation.clickable
+import androidx.compose.foundation.layout.*
+import androidx.compose.foundation.text.KeyboardOptions
+import androidx.compose.material.*
+import androidx.compose.material.icons.Icons
+import androidx.compose.material.icons.filled.ArrowDropDown
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.MutableState
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.setValue
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Alignment.Companion.CenterHorizontally
+import androidx.compose.ui.Alignment.Companion.CenterVertically
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.text.TextStyle
+import androidx.compose.ui.platform.ComposeView
+import androidx.compose.ui.text.input.KeyboardType
+import androidx.compose.ui.text.style.TextAlign
+import androidx.compose.ui.unit.dp
+import androidx.compose.ui.unit.sp
+import androidx.fragment.app.Fragment
+import com.github.scribeWizTeam.scribewiz.Util.RecordingParameters
+import com.github.scribeWizTeam.scribewiz.Activities.NavigationActivity
+import com.github.scribeWizTeam.scribewiz.PermissionsManager
+import com.github.scribeWizTeam.scribewiz.NotesStorageManager
+import com.github.scribeWizTeam.scribewiz.R
+import java.io.File
+import java.io.FileOutputStream
+import java.io.IOException
+import java.nio.ByteBuffer
+
+import kotlin.math.*
+
+
+class RecParameterFragment(contentLayoutId: Int,
+                           val navActivity: NavigationActivity) : Fragment(contentLayoutId) {
+
+    companion object {
+        private val TONALITIES_NAMES = listOf(
+            "A", "B", "C", "D", "E", "F", "G",
+            "Ab", "Bb", "Db", "Eb", "F#",
+            "a", "b", "c", "d", "e", "f", "g",
+            "bb", "c#", "d#", "f#", "g#"
+        )
+
+        private val TONALITIES_FIFTHS = listOf(
+            3, 5, 0, 2, 4, -1, 1,
+            -4, -2, -5, -3, 6,
+            0, 2, -3, -1, 1, -4, -2,
+            -5, 4, 6, 3, 5
+        )
+
+        private val KEY_NAMES = listOf("G", "F")
+        private val KEY_VALUES = listOf(true, false)
+
+        private val BEAT_TYPES = listOf(1, 2, 4, 8, 16)
+
+    }
+
+    private var recording_parameters = RecordingParameters()
+
+    override fun onCreateView(
```

Comment:
This one is maybe a bit too long and could be modularized with compose function, one per major component maybe